auth/service/api/v1/profile.go

```diff
@@ -103,64 +104,58 @@ func (r *Router) GetUsersWithProfiles(res rest.ResponseWriter, req *rest.Request
-		// Does this mean all users should already be migrated
-		// to keycloak before this call? Or should UserAccessor have a "fallback" like shoreline's legacy mongodb repo?
-		sharedUser, err := r.UserAccessor().FindUserById(ctx, userID)
-		if stdErrs.Is(err, user.ErrUserNotFound) || sharedUser == nil {
-			// According to seagull code, "It's possible for a user profile to be deleted before the sharing permissions", so we can ignore if user or profile not found.
-			continue
-		}
-		if err != nil {
-			responder.InternalServerError(err)
-			return
-		}
-		if !userMatchesQueryOnUser(sharedUser, filter) {
-			continue
-		}
-		profile, err := r.getProfile(ctx, userID)
-		if stdErrs.Is(err, user.ErrUserProfileNotFound) || profile == nil {
-			continue
-		}
-		if err != nil {
-			r.handleProfileErr(responder, err)
-			return
-		}
-		trustorPerms := trustPerms.TrustorPermissions
-		if trustorPerms == nil || len(*trustorPerms) == 0 {
-			profile = profile.ClearPatientInfo()
-		} else {
-
-			if trustorPerms.HasAny(permission.Custodian, permission.Read, permission.Write) {
-				// TODO: need to read seagull.value.settings
+	group, ctx := errgroup.WithContext(ctx)
+	group.SetLimit(20) // do up to 20 concurrent requests like seagull did
+	for userID, trustPerms := range mergedUserPerms {
+		userID, trustPerms := userID, trustPerms
+		group.Go(func() error {
+			sharedUser, err := r.UserAccessor().FindUserById(ctx, userID)
+			if stdErrs.Is(err, user.ErrUserNotFound) || sharedUser == nil {
+				// According to seagull code, "It's possible for a user profile to be deleted before the sharing permissions", so we can ignore if user or profile not found.
+				return nil
 			}
-			if trustorPerms.Has(permission.Custodian) {
-				// TODO: need to read seagull.value.preferences
+			if err != nil {
+				return err
 			}
-		}
-		sharedUser.Profile = profile
-		sharedUser.TrusteePermissions = trustPerms.TrusteePermissions
-		sharedUser.TrustorPermissions = trustPerms.TrustorPermissions
-		// Seems no sharedUser.Sanitize call to filter out "protected" fields in seagull except sanitizeUser to remove "passwordExists" field - which doesn't exist in current platform/user.User
-		matchedUser := userMatchingQuery(sharedUser, filter)
-		if matchedUser != nil {
-			results = append(results, matchedUser)
-		}
+			profile, err := r.getProfile(ctx, userID)
+			if stdErrs.Is(err, user.ErrUserProfileNotFound) || profile == nil {
+				return nil
+			}
+			if err != nil {
+				return err
+			}
+			trustorPerms := trustPerms.TrustorPermissions
+			if trustorPerms == nil || len(*trustorPerms) == 0 {
+				profile = profile.ClearPatientInfo()
+			} else {
+				if trustorPerms.HasAny(permission.Custodian, permission.Read, permission.Write) {
+					// TODO: need to read seagull.value.settings - confirm this is actually used
+				}
+				if trustorPerms.Has(permission.Custodian) {
+					// TODO: need to read seagull.value.preferences - confirm this is actually used
```

Comment:
Preferences are not part of the profile object so probably there's no need to do anything?